pkg/ovsdb/transact.go

```diff
@@ -1556,52 +1513,77 @@ func preWait(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.Op
 	return etcdGetByWhere(txn, ovsOp, ovsResult)
 }
 
-func doWaitDummy(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.OperationResult) error {
-	return nil
-}
-
+/* wait */
 func doWait(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.OperationResult) error {
-	rows := []map[string]interface{}{}
+	if ovsOp.Table == nil {
+		klog.Errorf("missing table parameter")
+		return errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if ovsOp.Rows == nil {
+		klog.Errorf("missing rows parameter")
+		return errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if ovsOp.Columns == nil {
+		klog.Errorf("missing columns parameter")
+		return errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if ovsOp.Until == nil {
+		klog.Errorf("missing until parameter")
+		return errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
```

Comment:
should you check that ovsOp.Where is not nil ? ( it is a required parameter too)

Reply:
I assume if it does not exist that means everything